src/serac/physics/utilities/finite_element_dual.hpp

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) 2019-2021, Lawrence Livermore National Security, LLC and
+// other Serac Project Developers. See the top-level LICENSE file for
+// details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+
+/**
+ * @file finite_element_dual.hpp
+ *
+ * @brief This contains a class that represents the dual of a finite element vector space, i.e. the space of residuals
+ * and sensitivities.
+ */
+
+#pragma once
+
+#include <memory>
+
+#include "mfem.hpp"
+
+#include "serac/physics/utilities/finite_element_vector.hpp"
+
+namespace serac {
+
+/**
+ * @brief Class for encapsulating the dual vector space of a finite element space (i.e. the
+ * space of linear forms as applied to a specific basis set)
+ */
+class FiniteElementDual : public FiniteElementVector {
+public:
+  /**
+   * Main constructor for building a new finite element dual
+   * @param[in] mesh The problem mesh (object does not take ownership)
+   * @param[in] options The options specified, namely those relating to the order of the problem,
+   * the dimension of the FESpace, the type of FEColl, the DOF ordering that should be used,
+   * and the name of the field
+   */
+  FiniteElementDual(mfem::ParMesh& mesh, Options&& options = {})
+      : FiniteElementVector(mesh, std::forward<FiniteElementVector::Options>(options)){};
+
+  /**
+   * @brief Minimal constructor for a FiniteElementDual given an already-existing field
+   * @param[in] mesh The problem mesh (object does not take ownership)
+   * @param[in] gf The field for the dual to create (object does not take ownership)
+   * @param[in] name The name of the field
+   */
+  FiniteElementDual(mfem::ParMesh& mesh, mfem::ParGridFunction& gf, const std::string& name = "")
+      : FiniteElementVector(mesh, gf, name){};
+
+  /**
+   * @brief Minimal constructor for a FiniteElementDual given a finite element space
+   * @param[in] mesh The problem mesh (object does not take ownership)
+   * @param[in] space The space to use for the finite element dual. This space is deep copied into the new FE state
+   * @param[in] name The name of the field
+   */
+  FiniteElementDual(mfem::ParMesh& mesh, mfem::ParFiniteElementSpace& space, const std::string& name = "")
+      : FiniteElementVector(mesh, space, name){};
```

Comment:
I think these might be replaceable by 
`using FiniteElementVector::FiniteElementVector;`

see https://godbolt.org/z/xjh9enTnd

Reply:
TIL! Thanks!